pkg/controller/v1alpha1/autoscaler/autoscaler_controller.go

```diff
@@ -0,0 +1,129 @@
+/*
+
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package controllers
+
+import (
+	"context"
+	"flag"
+	"path/filepath"
+	"reflect"
+	"time"
+
+	"github.com/crossplane/crossplane-runtime/pkg/event"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+
+	"k8s.io/client-go/util/homedir"
+
+	kedav1alpha1 "github.com/kedacore/keda/api/v1alpha1"
+
+	"k8s.io/client-go/tools/clientcmd"
+
+	"github.com/go-logr/logr"
+	"github.com/oam-dev/kubevela/api/v1alpha1"
+	"k8s.io/apimachinery/pkg/runtime"
+	restclient "k8s.io/client-go/rest"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	SpecWarningTargetWorkloadNotSet = "Spec.targetWorkload is not set"
+	SpecWarningStartAtTimeFormat    = "startAt is not in the right format, which should be like `12:01`"
+)
+
+var (
+	scaledObjectKind       = reflect.TypeOf(kedav1alpha1.ScaledObject{}).Name()
+	scaledObjectAPIVersion = "keda.k8s.io/v1alpha1"
```

Comment:
why not use `SchemeGroupVersion` in https://github.com/kedacore/keda/blob/v2/api/v1alpha1/groupversion_info.go#L33 ?